File: scripts/update-excel.js
#!/usr/bin/env node

/**
 * Update Excel Cell Script
 * Usage: node update-excel.js <filePath> <cellAddress> <cellValue>
 * Example: node update-excel.js /path/to/file.xlsx E8 "4/19/2026"
 */

const xlsx = require('xlsx');
const fs = require('fs');
const path = require('path');

const args = process.argv.slice(2);

if (args.length < 3) {
  console.error('Usage: node update-excel.js <filePath> <cellAddress> <cellValue>');
  process.exit(1);
}

const [filePath, cellAddress, cellValue] = args;

try {
  // Validate file exists
  if (!fs.existsSync(filePath)) {
    throw new Error(`File not found: ${filePath}`);
  }

  // Read the workbook
  const workbook = xlsx.readFile(filePath);
  const worksheet = workbook.Sheets[workbook.SheetNames[0]];

  // Update the cell
  worksheet[cellAddress] = { v: cellValue, t: 's' };

  // Write back to file
  xlsx.writeFile(workbook, filePath);

  console.log(JSON.stringify({
    success: true,
    message: `Updated ${cellAddress} with value: ${cellValue}`,
    filePath: filePath,
    cellAddress: cellAddress,
    cellValue: cellValue
  }));

  process.exit(0);
} catch (error) {
  console.error(JSON.stringify({
    success: false,
    error: error.message
  }));
  process.exit(1);
}

File: scripts/update-excel.py
#!/usr/bin/env python3

"""
Update Excel Cell Script
Usage: python3 update-excel.py <filePath> <cellAddress> <cellValue>
Example: python3 update-excel.py /path/to/file.xlsx E8 "4/19/2026"
"""

import sys
import json
from openpyxl import load_workbook

def update_excel_cell(file_path, cell_address, cell_value):
    try:
        # Load the workbook
        wb = load_workbook(file_path)
        ws = wb.active
        
        # Update the cell
        ws[cell_address] = cell_value
        
        # Save the workbook
        wb.save(file_path)
        
        return {
            "success": True,
            "message": f"Updated {cell_address} with value: {cell_value}",
            "filePath": file_path,
            "cellAddress": cell_address,
            "cellValue": cell_value
        }
    except Exception as error:
        return {
            "success": False,
            "error": str(error)
        }

if __name__ == "__main__":
    if len(sys.argv) < 4:
        print(json.dumps({
            "success": False,
            "error": "Usage: python3 update-excel.py <filePath> <cellAddress> <cellValue>"
        }))
        sys.exit(1)
    
    file_path = sys.argv[1]
    cell_address = sys.argv[2]
    cell_value = sys.argv[3]
    
    result = update_excel_cell(file_path, cell_address, cell_value)
    print(json.dumps(result))
    
    if result["success"]:
        sys.exit(0)
    else:
        sys.exit(1)
